### ISSUE TITLE
Support multi-row INSERT .. RETURNING for Oracle using PL/SQL FORALL .. INSERT .. RETURNING .. BULK COLLECT INTO

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#5863]
This is how we can `INSERT .. RETURNING` in Oracle:

```sql
SET SERVEROUTPUT ON
DECLARE
  -- Input bind value(s)
  in_c2_p sys.ora_mining_number_nt := sys.ora_mining_number_nt(12, 13);
  
  in_c2  dbms_sql.varchar2_table;
  out_c1 dbms_sql.number_table;
  out_c2 dbms_sql.varchar2_table;
  
  cur_c1 sys_refcursor;
  cur_c2 sys_refcursor;
BEGIN
  FOR i IN 1 .. in_c2_p.COUNT LOOP
    in_c2(i) := in_c2_p(i);
  END LOOP;
  
  -- The FORALL loop inserts individual rows and fetches all results
  FORALL i IN 1 .. in_c2.COUNT
    INSERT INTO t_identity_pk (val)
    VALUES (in_c2(i))
    RETURNING id, val BULK COLLECT INTO out_c1, out_c2;
  
  -- Output bind value(s)
  OPEN cur_c1 FOR SELECT * FROM TABLE (out_c1);
  dbms_sql.return_result(cur_c1);
  
  OPEN cur_c2 FOR SELECT * FROM TABLE (out_c2);
  dbms_sql.return_result(cur_c2);
END;
/
```

Some more inspiration here: http://www.oracle-developer.net/display.php?id=413

**Comment by lukaseder** [comment]
This needs much more research, also in terms of performance. We'll postpone this to 3.11

**Comment by lukaseder** [comment]
This emulation has already been implemented for `UPDATE` and `DELETE` through #5191. Should be an easy fix for `INSERT`
